src/main/java/io/hops/util/FeaturestoreRestClient.java

```diff
@@ -488,5 +490,148 @@ public static FeaturestoreJdbcConnectorDTO getOnlineFeaturestoreJdbcConnectorRes
     JSONObject onlineFeaturestoreConnector = new JSONObject(responseEntity);
     return FeaturestoreHelper.parseJdbcConnectorJson(onlineFeaturestoreConnector);
   }
+  
+  /**
+   * Makes a REST call to Hopsworks to attach extended metadata to a featuregroup
+   * @param featuregroupName the feature group name
+   * @param featurestore the feature store
+   * @param featuregroupVersion the feature group version
+   * @param name the extended attribute name
+   * @param value the extended attribute value
+   * @throws FeaturestoreNotFound
+   * @throws JAXBException
+   * @throws FeaturegroupDoesNotExistError
+   * @throws FeaturegroupMetadataError
+   */
+  public static void addMetadata(String featuregroupName, String featurestore,
```

Comment:
Would it be possible to add multiple values at ones? Like passing a dictionary? The same when removing. It would also be nice if we would add multiple sets of keys and values in an atomic way in the backend to avoid inconsistency.

Reply:
Yes, it is possible, however, i had a short discussion with @tkakantousis while writing the REST API if i should go with the bulk approach or leave it for now as a for loop on the client side. From my perspective, i prefer the bulk approach to send a dictionary, rather than doing a for loop in the client side. I'm using the file system operations to add/get/remove extended attributes attached to a file. With the current API, consistency is guaranteed for each key value pair not for the whole set of keys unless we add another API on the file system side to support adding a dictionary.

Reply:
OK it's not worth changing the filesystem at this point and I'm OK with exposing the actual behavior via the API.

Reply:
Since we cannot provide atomic commit for multiple <k,v> pairs and we do not have a `/bulk` endpoint at the moment, suggested solution in this PR is sufficient. Otherwise, we need to implement a proper bulk endpoint in Hopsworks that access resources to be bulk loaded so it has to be generic, not only for xattrs. That however goes beyond the scope of this PR and we want to avoid another blocker.